src/resolve/resolved-conf.c

```diff
@@ -413,3 +413,47 @@ int manager_parse_config_file(Manager *m) {
 #endif
         return 0;
 }
+
+int config_parse_refuse_record_types(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Manager *m = ASSERT_PTR(userdata);
```

Comment:
Then, please take a pointer to `Set` from `data`, rather than `Manager` from `userdata`:
```c
Set *types = ASSERT_PTR(data);
```